readme.md

```diff
@@ -192,16 +192,22 @@ Yields:
 This package exports the identifiers `createStarryNight`, `common`, and `all`.
 There is no default export.
 
-### `createStarryNight(grammars)`
+### `createStarryNight(grammars[, options])`
 
 Create a `StarryNight` that can highlight things based on the given `grammars`.
 This is async to facilitate async loading and registering, which is currently
-only used for WASM.
+only used for WASM.  
+Use the optional `Options` object to pass functions that resolve to a `URL`
+pointing to a `vscode-oniguruma` WASM binary.
+
+*Note: The URL must result in a version 1 `onig.wasm` from `vscode-oniguruma`*
```

Comment:
```suggestion
```

Better to document `Options` and `GetOnigurumaUrl` separately.
Perhaps right before `` ### `starryNight.flagToScope(flag)` ``, in this section, as a h4?

I’d recommend something along the lines of:

```markdown
#### `Options`

Configuration (optional).

###### `options.getOnigurumaUrlFetch`

Get a URL to the oniguruma WASM, typically used in browsers (`GetOnigurumaUrl`,
optional).

###### `options.getOnigurumaUrlFs`

Get a URL to the oniguruma WASM, typically used in Node.js (`GetOnigurumaUrl`,
optional).

#### `GetOnigurumaUrl`

Function to get a URL to the oniguruma WASM.

> 👉 **Note**: this must currently result in a version 1 URL of
> `onig.wasm` from `vscode-oniguruma`.

###### Returns

URL object to a WASM binary (`URL` or `Promise<URL>`).
```

Reply:
added in 63950af014610c308790850465a65168d84ac5e0